src/main/kotlin/com/pubnub/api/models/consumer/objects/PNSortKey.kt

```diff
@@ -4,15 +4,15 @@ enum class PNKey(internal val fieldName: String) {
     ID("id"), NAME("name"), UPDATED("updated");
 }
 
-sealed class PNSortKey(
-    private val key: PNKey,
-    private val dir: String = "asc"
-) {
+open class PNSortKey(
```

Comment:
Also I think we might get away with still just this class without copying it multiple times for different type of `keys`. The plan would be to have it like this:

```

interface SortField {
    val fieldName: String
}

sealed class PNSortKey<T : SortField>(
    internal val key: T,
    internal val dir: String = "asc"
) {
    class PNAsc<T : SortField>(key: T) : PNSortKey<T>(key = key, dir = "asc")
    class PNDesc<T : SortField>(key: T) : PNSortKey<T>(key = key, dir = "desc")

    fun toSortParameter(): String {
        return key.fieldName + ":" + dir
    }

    companion object {
        fun asc(key: PNKey): PNSortKey<PNKey> {
            return PNAsc(key)
        }

        fun desc(key: PNKey): PNSortKey<PNKey> {
            return PNDesc(key)
        }
    }
}

```

Then `SortBase` can go away because it's not needed after all.

Reply:
Ok, make sense.